### PR TITLE
add DEX contract explanations and put a guide in README

### LINKED ISSUE #9: Write Get Started guide in README
- don't assume users know how to do complex setup scenario
- try to make instruction as detailed as possible
- give references to necessary related setup instruction (google for once)
- the manual should be accessible to anyone new to Scala (This is especially important if we want to target large audience. We don't use complex Scala in the contracts, and newbies should feel comfortable is the contracts, but basic Scala setup is usually an obstacle. Let's help with this)

### LINKED ISSUE #10: Provide detailed documentation of the contracts used in DEX
The documentation should explain the logic on the contract and usage context.
Since we are referring it from DEX blog post, the description should be self sufficient. 
In particular it should reference back to the ergo-tool commands where the contracts are used.

### PR DESCRIPTION
**Pull request by greenhat**
Close #9 
Close #10

### CODE REVIEW

File: verified-contracts/src/main/scala/org/ergoplatform/contracts/AssetsAtomicExchange.scala [hunk 1]
```diff
@@ -9,50 +9,62 @@ import scala.language.{implicitConversions, postfixOps}
 
 sealed abstract class AssetsAtomicExchange extends SigmaContract {
 
-  /** Buyer's contract for DEX
+  /** DEX contract for buy order
+    * Bid order to buy a token specified by token id and amount for a price specified in a box value that
+    * will be protected by this contract. This box can be spend in two ways:
+    * 1. Buyer wants to cancel the order and return the coins (buyerPk is enough).
+    * 2. Seller creates a box with a specified token id and amount,
+    * puts this buy order box id in R4, and make it claimable by buyerPk.
```
Comment (aslesarenko) [inline verified-contracts/src/main/scala/org/ergoplatform/contracts/AssetsAtomicExchange.scala]: Case 2) has incorrect description. 
It is not seller who create the spending transaction (it doesn't matter who create it).
Note, that buyerPk is not necessary on this spending path.

File: README.md [hunk 2]
```diff
@@ -1,2 +1,25 @@
...
+## Prerequisites:
+- Install Z3 SMT solver from https://github.com/Z3Prover/z3
+
+## How to add a new contract
+
+### Create a method for the contract
+Subclass `SigmaContract` in the `verified-contracts` project and put a contract code in a method. The first parameter has to be `ctx: Context`, and subsequent parameters may be contract parameters. The return value has to be `SigmaProp`. Make the first line of the contract code `import ctx._` to improve readability.
+
+### Write contract code in the method.
+See [DEX buy order](http://github.com/ScorexFoundation/sigmastate-interpreter/blob/346717a7d6af45c685b617ccea7babcf39d49939/contract-verification/src/main/scala/sigmastate/verification/contract/AssetsAtomicExchange.scala#L12-L32) for an example.
```
Comment (aslesarenko) [inline README.md]: The url is not correct.

File: README.md [hunk 3]
```diff
@@ -1,2 +1,25 @@
...
+
+### Contract compilation
+Create a subclass (object) of the class with contract code to make "instance" method to compile the contract's code.
+It'll invoke the compiler (macros) and returns a compiled contract with embedded contract parameters. Create a method with parameters from the contract (without the `Context` parameter) and invoke `ErgoContractCompiler.compile`. See [DEX buy order](http://github.com/ScorexFoundation/sigmastate-interpreter/blob/346717a7d6af45c685b617ccea7babcf39d49939/contract-verification/src/main/scala/sigmastate/verification/contract/AssetsAtomicExchange.scala#L116-L122) for an example.
+Mark this method with `@ignore` annotation to hide it from Stainless.
```
Comment (aslesarenko) [inline README.md:17]: Url.

File: README.md [hunk 4]
```diff
@@ -1,2 +1,25 @@
...
+
+### Contract compilation
+Create a subclass (object) of the class with contract code to make "instance" method to compile the contract's code.
+It'll invoke the compiler (macros) and returns a compiled contract with embedded contract parameters. Create a method with parameters from the contract (without the `Context` parameter) and invoke `ErgoContractCompiler.compile`. See [DEX buy order](http://github.com/ScorexFoundation/sigmastate-interpreter/blob/346717a7d6af45c685b617ccea7babcf39d49939/contract-verification/src/main/scala/sigmastate/verification/contract/AssetsAtomicExchange.scala#L116-L122) for an example.
+Mark this method with `@ignore` annotation to hide it from Stainless. 
+
+### How to use compiled contract
+Call the "instance" method in another module/project and it'll return 'ErgoContract'(compiled contract).
+Call `ErgoContract.scalaFunc` to run the contract with given `Context`. See [DEX buy order](http://github.com/ScorexFoundation/sigmastate-interpreter/blob/346717a7d6af45c685b617ccea7babcf39d49939/sigma-dsl-compiler-macros-playground/src/test/scala/sigmastate/verification/test/AssetsAtomicExchangeCompilationTest.scala#L177-L220) for an example.
+
```
Comment (aslesarenko) [inline README.md:22]: This urls also lead to sigma, why?

**Review by aslesarenko — CHANGES_REQUESTED** [review]

File: verified-contracts/src/main/scala/org/ergoplatform/contracts/AssetsAtomicExchange.scala (hunk 1 above)
Comment (greenhat) [reply verified-contracts/src/main/scala/org/ergoplatform/contracts/AssetsAtomicExchange.scala]: Updated.

File: README.md (hunk 2 above)
Comment (greenhat) [reply README.md]: Fixed.

File: README.md (hunk 3 above)
Comment (greenhat) [reply README.md:17]: Fixed.

File: README.md (hunk 4 above)
Comment (greenhat) [reply README.md:22]: From my old manual written in sigma times. Thanks for the catch! Fixed.

**Review by aslesarenko — APPROVED** [review]

**Pull request merged**